sycl/doc/extensions/proposed/sycl_ext_codeplay_cuda_cluster_group.asciidoc

```diff
@@ -0,0 +1,421 @@
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not version-ed, so the
+ feature-test macro always has this value.
+|===
+
+
+=== Extension to `enum class aspect`
+
+[source]
+----
+namespace sycl {
+enum class aspect {
+  ...
+  ext_codeplay_cuda_cluster_group
+}
+}
+----
+
+A device requires the `ext_codeplay_cuda_cluster_group` aspect to
+support launching a kernel with the `cluster_size` property defined in the
+following section.
+
+
+=== Launching a kernel with a `cluster_group`
+
+Because of the special scheduling guarantees associated with a cluster launch,
+the backend must know which kernel would be using this feature. Thus, this 
+proposal introduces a new launch property called as `cluster_size` that will
+contain the cluster size as a number of work-groups.
+ 
+[source,c++]
+----
+namespace sycl::ext::codeplay::experimental::cuda {
+/**
+* Dim Dimensionality of the launch
+* size sycl::range specifying the number of work-groups in the cluster
+  in each dimension.
+*/
+template <int Dim>
+struct cluster_size {
+  cluster_size(const sycl::range<Dim>& size);
+  sycl::range<Dim> get_cluster_size();
+  ...
+};
+using cluster_size_key = cluster_size;
+} // namespace sycl::ext::codeplay::experimental::cuda
+----
+
+The property list can the be constructed as follows - 
+
+[source,c++]
+----
+properties  cluster_launch_property{cluster_size({1, 2, 1})};
+----
+
+Note that the total number of work-groups in the kernel must be a multiple of
+the cluster size in each dimension.
+
+The launch functions introduced in `sycl_ext_oneapi_enqueue_functions` can then
+be used to launch the kernel with the `cluster_size` property.
+
+
+=== Querying Maximum Cluster Size
+
+To query the maximum supported cluster size, this proposal adds a new device
+query, `max_cluster_group_size`, which returns the maximum possible number of 
+work-groups present inside the cluster.
+
+[source, c++]
+----
+size_t max_cluster_size = 
+  device.get_info<
+    ext::codeplay::experimental::cuda::info::device::max_cluster_group_size>();
+----
+[%header,cols="10,5,5"]
+|===
+|Device descriptor
+|Return Type
+|Description
+
+|`ext::codeplay::experimental::cuda::info::device::max_cluster_group_size`
+|size_t
+|Returns the maximum possible number of work-groups that can constitute a 
+cluster-group
+|===
+
+
+=== Accessing the Cluster Group From Device Code
+
+Building upon the group hierarchy in SYCL, this proposal adds another level
+above group (for work-groups), to be called as `cluster-group`, which 
+represents a collection of work-groups and will be accessible via the `nd_item`
+class, via a member function to be introduced called `ext_codeplay_cuda_get_cluster_group()`.
+
+
+[%header,cols="10,5"]
+|===
+|Method
+|Description
+
+|`cluster_group<Dim> nd_item::ext_codeplay_cuda_get_cluster_group()`
+|Returns the constituent `cluster_group` in the kernel, representing this
+`cluster_group`'s overall position in the `nd_range`
```

Comment:
```suggestion
`cluster_group` object's overall position in the `nd_range`
```

The text you had was not formatting correctly.  It's usually awkward to make a C++ type name possessive (or plural).  Instead, I usually find a way to rewrite the sentence to avoid this.  In addition, this proposed text fixes the formatting problem.